content/purgatorio/now-go-build/transfer-learning.md

```diff
@@ -1,5 +1,111 @@
+- Robustness: you make the model more "robust" to real-world low-quality data  
+
+So, **the first thing to do** every time you frame a new problem, is to ask yourself:
+
+**Can I leverage Transfer Learning in solving this problem?**
+
+
+## How Does Transfer Learning Work
+
+The practice of Transfer Learning allows reusing most of the parameters (weights) of a neural network previously trained on a problem similar to the one we have to solve, dwelling on the training only of the last layers that are usually those dedicated to the classification and/or regression of the features obtained with the previous layers.
+
+This allows us to obtain two key results:
+
+- reuse of the behavior of a network already trained to effectively **extract features** from input data
+- limit processing to a significantly **smaller number of parameters** (corresponding to the last layers)
+
+For example, if we were to classify apple varieties from an image, we could start by using a neural network already trained to classify images of planes, cars, dogs, cats, eggs, etc... This is because of the greater variety of training dataset ensures a better ability to extract features of various kinds from images.
+
+Of the _pre-trained neural network_, we would keep only the initial layers, going to redefine only the last layers of classification.
+
+The reused layers would be labeled as "read-only", in order to calculate only the parameters corresponding to the last layers, speeding up the training time, reducing the processing power required, and generally improving the accuracy.
+
+In general, the set of layers that are reused by a pre-trained network is called a backbone or feature-extractor.
+
+The practice of applying Transfer Learning theory to real-world models is called "fine-tuning".
+
+To learn more about Transfer Learning:
+
+- [A Comprehensive Survey on Transfer Learning](https://arxiv.org/abs/1911.02685?utm_source=feedburner&utm_medium=feed&utm_campaign=Feed%253A+arxiv%252FQSXk+%2528ExcitingAds%2521+cs+updates+on+arXiv.org%2529)
+- [How transferable are features in deep neural networks?](https://papers.nips.cc/paper/5347-how-transferable-are-features-in-deep-neural-networks.pdf)
+- [Using Pre-Training Can Improve Model Robustness and Uncertainty](https://arxiv.org/abs/1901.09960)
+- [What makes ImageNet good for transfer learning?](https://arxiv.org/abs/1608.08614)
+
+## Where to Find Pretrained Models
+
+Having clear in mind how useful it is to be able to use pre-trained templates to solve your problem, let's now see where on the Internet you can find these templates, plus some guides and tutorials to do the _fine tune_ them.  
+
+Some general purposes places where you fill find pre-trained models are:
+
+- [Model Zoo](https://modelzoo.co/)
+- [TensorFlow Hub](https://www.tensorflow.org/hub)
+- [Papers with Code](https://paperswithcode.com/)
+- [Keras Pre-Trained Models](https://keras.rstudio.com/articles/applications.html)
+
```

Comment:
A link https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/README-details.md#models to find some models

Reply:
I think there are already links enough (redundant in this case)